### ISSUE TITLE
Streaming RNNT ASR with context 

### ISSUE THREAD

**Issue by robertveenhoven** [NVIDIA/NeMo#5891]
Im trying to implement streaming ASR from microphone with PyAudio. However, Im a bit stuck on which functions to use exactly.
Im now using pyaudio with stream_callback as provided by Online_ASR_Microphone_Demo.ipynb, but Im stuck in how to progress further.

I found the script of speech_to_text_cache_aware_streaming_infer.py, which uses conformer_stream_step() in combination with a model trained specifically for streaming, but there are also a lot of functions in streaming_utils.py as StreamingFeatureBufferer and FrameASR to provide buffers and transcribe from there, however in my understanding the transcription wouldnt use context.

Im a bit stuck on which functions I should for streaming ASR with the best possible result. Can I directly use the chunks from the audio in the conformer_stream_step() or do I need something like StreamingFeatureBufferer in between?

Could you provide an overview on which functions I should use to provide such a pipeline? And should I train a streaming model? Or would a full context model work fine.

**Comment by VahidooX** [comment]
We have not released any checkpoints trained with cache-aware streaming yet. We are going to release some in the following months. In the meantime, I recommend to use the buffered streaming with the offline pretrained checkpoints. 

The current offline checkpoints released publicly on NGC has been trained on large datasets and not easy to beat them. Unless you are working on a model for a language other than English. The offline model can be used with cache-aware streaming inference but accuracy would be worse than buffered streaming.

**Comment by robertveenhoven** [comment]
We do train on Dutch and also finetuned the conformer-transducer large on it. Can we use that checkpoint to finetune for streaming asr providing the config file in the cache-aware streaming folder?

And then which functions from streaming_utils should we use to move from streaming an audio file to streaming real time using a microphone with pyaudio?

**Comment by robertveenhoven** [comment]
Any updates? @VahidooX

**Comment by VahidooX** [comment]
We have released some cache-aware FastConformer models:
https://registry.ngc.nvidia.com/orgs/nvidian/teams/ac-aiapps/models/stt_en_fastconformer_hybrid_large_streaming_1040ms
https://registry.ngc.nvidia.com/orgs/nvidian/teams/ac-aiapps/models/stt_en_fastconformer_hybrid_large_streaming_480ms
https://registry.ngc.nvidia.com/orgs/nvidian/teams/ac-aiapps/models/stt_en_fastconformer_hybrid_large_streaming_80ms
